### ISSUE TITLE
Insecure/obsolete default signature algorithm

### ISSUE THREAD

**Issue by simongellis-attentive** [SAML-Toolkits/java-saml#415]
If you don't provide a value for `onelogin.saml2.security.signature_algorithm`, this library defaults to [signing requests with RSA-SHA1](https://github.com/SAML-Toolkits/java-saml/blob/baee4e0a0c6ab6411034d2413e837f4403f1a11f/core/src/main/java/com/onelogin/saml2/settings/Saml2Settings.java#L79).

SHA-1 as a signature algorithm is obsolete. [Wikipedia says](https://en.wikipedia.org/wiki/SHA-1) that it is
> recommended to remove SHA-1 from products as soon as possible and instead use [SHA-2](https://en.wikipedia.org/wiki/SHA-2) or [SHA-3](https://en.wikipedia.org/wiki/SHA-3). Replacing SHA-1 is urgent where it is used for [digital signatures](https://en.wikipedia.org/wiki/Digital_signatures).

It would probably make more sense to default to SHA-256 instead.

**Comment by haavar** [comment]
I agree this is problematic. I think the defaults should change. I discovered that our configuration signed the authN requests with SHA1, and it seems like ADFS will need to sign the authN response with SHA1 when the request is signed with SHA1. 

I think the default should also change to not allow SHA1 signatures in the response. Someone spoofing a response worries me more than someone spoofing a request.
